pandas/tests/io/test_parquet.py

```diff
@@ -0,0 +1,232 @@
+
+        self.check_round_trip(df, pa)
+
+    def test_duplicate_columns(self, pa):
+
+        # not currently able to handle duplicate columns
+        df = pd.DataFrame(np.arange(12).reshape(4, 3),
+                          columns=list('aaa')).copy()
+        self.check_error_on_write(df, pa, ValueError)
+
+    def test_unsupported(self, pa):
+        from pyarrow.error import ArrowException
+
+        # period
+        df = pd.DataFrame({'a': pd.period_range('2013', freq='M', periods=3)})
+        self.check_error_on_write(df, pa, ArrowException)
+
+        # categorical
+        df = pd.DataFrame({'a': pd.Categorical(list('abc'))})
+        self.check_error_on_write(df, pa, ArrowException)
+
+        # date_range
+        df = pd.DataFrame({'a': pd.date_range('20130101', periods=3)})
+        self.check_error_on_write(df, pa, ArrowException)
+
+        # date_range w/tz
+        df = pd.DataFrame({'a': pd.date_range('20130101', periods=3,
+                                              tz='US/Eastern')})
+        self.check_error_on_write(df, pa, ArrowException)
+
+    def test_mixed(self, pa):
+        # mixed python objects are returned as None ATM
+        df = pd.DataFrame({'a': ['a', 1, 2.0]})
+        expected = pd.DataFrame({'a': ['a', None, None]})
+
+        self.check_round_trip(df, pa, expected)
+
+
+class TestParquetFastParquet(Base):
+
+    def test_basic(self, fp):
+
+        df = pd.DataFrame({'string': list('abc'),
+                           'int': list(range(1, 4)),
+                           'uint': np.arange(3, 6).astype('u1'),
+                           'float': np.arange(4.0, 7.0, dtype='float64'),
+                           'bool': [True, False, True],
+                           'datetime': pd.date_range('20130101', periods=3),
+                           })
+
+        self.check_round_trip(df, fp)
+
+    @pytest.mark.skip(reason="not supported")
+    def test_duplicate_columns(self, fp):
+
+        # not currently able to handle duplicate columns
+        df = pd.DataFrame(np.arange(12).reshape(4, 3),
+                          columns=list('aaa')).copy()
+        self.check_error_on_write(df, fp, ValueError)
+
+    def test_unsupported(self, fp):
+
+        # period
+        df = pd.DataFrame({'a': pd.period_range('2013', freq='M', periods=3)})
+        self.check_error_on_write(df, fp, ValueError)
+
+        # mixed
+        df = pd.DataFrame({'a': ['a', 1, 2.0]})
+        self.check_error_on_write(df, fp, ValueError)
+
+    def test_categorical(self, fp):
+        # returns as object
+        df = pd.DataFrame({'a': pd.Categorical(list('abc'))})
+        self.check_round_trip(df, fp, df.assign(a=df.a.astype(object)))
+
+    def test_datetime_tz(self, fp):
+        # doesn't preserve tz
+        df = pd.DataFrame({'a': pd.date_range('20130101', periods=3,
+                                              tz='US/Eastern')})
+
+        # warns on the coercion
+        with catch_warnings(record=True):
```

Comment:
We should create a spec document for "Python metadata" (that we can implement in either library) to store as key-value metadata so that we can persist anything that does not fit into the format as is